dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -77,4 +106,198 @@ private FieldSpec bindEndpointChannel(EndpointDefinition endpoint) {
+        //
+        //     @Override
+        //     public CodeBlock visitMap(MapType value) {
+        //         throw new SafeIllegalStateException("Maps can not be query parameters");
+        //     }
+        //
+        //     @Override
+        //     public CodeBlock visitReference(com.palantir.conjure.spec.TypeName typeName) {
+        //         TypeDefinition typeDef = typeNameResolver.resolve(typeName);
+        //         if (typeDef.accept(TypeDefinitionVisitor.IS_ALIAS)) {
+        //             return generatePlainSerializer(
+        //                     method,
+        //                     key,
+        //                     CodeBlock.of("$L.get()", argName),
+        //                     typeDef.accept(TypeDefinitionVisitor.ALIAS).getAlias());
+        //         } else if (typeDef.accept(TypeDefinitionVisitor.IS_ENUM)) {
+        //             return CodeBlock.of("$L.$L($S, $T.toString($L));", "_request", method, key, Objects.class,
+        // argName);
+        //         }
+        //         throw new IllegalStateException("Plain serialization can only be aliases and enums");
+        //     }
+        //
+        //     @Override
+        //     public CodeBlock visitExternal(ExternalReference value) {
+        //         // TODO(forozco): we could probably do something smarter than just calling toString
+        //         return CodeBlock.of("$L.$L($S, $T.toString($L));", "_request", method, key, Objects.class, argName);
+        //     }
+        //
+        //     @Override
+        //     public CodeBlock visitUnknown(String unknownType) {
+        //         throw new SafeIllegalStateException("Unknown param type", SafeArg.of("type", unknownType));
+        //     }
+        //
+        //     private CodeBlock visitCollection(Type itemType) {
+        //         CodeBlock elementVariable = CodeBlock.of("$LElement", argName);
+        //         return CodeBlock.builder()
+        //                 .beginControlFlow(
+        //                         "for ($T $L : $L)", parameterTypes.baseType(itemType), elementVariable, argName)
+        //                 .add(generatePlainSerializer(method, key, elementVariable, itemType))
+        //                 .endControlFlow()
+        //                 .build();
+        //     }
+        // });
+    }
+
+    // private static String getOptionalAccessor(Type type) {
+    //     if (type.accept(TypeVisitor.IS_PRIMITIVE)) {
+    //         PrimitiveType primitive = type.accept(TypeVisitor.PRIMITIVE);
+    //         if (primitive.equals(PrimitiveType.DOUBLE)) {
+    //             return "getAsDouble";
+    //         } else if (primitive.equals(PrimitiveType.INTEGER)) {
+    //             return "getAsInt";
+    //         }
+    //     }
+    //     return "get";
+    // }
+
+    private static final ImmutableMap<TypeName, String> PRIMITIVE_TO_TYPE_NAME =
+            ImmutableMap.copyOf(KeyedStream.stream(new ImmutableMap.Builder<Class<?>, String>()
```

Comment:
I feel like it would be good to generate that...

Reply:
Actually, this is not Jackson, this is for path/query/header params, so if someone is relying on a type that we were to later add here, they have likely wrapped it: it must've come from a library, and the library probably does not have the #valueOf -> String method that we would generate a binding to. Soo, I think it's ok to generate this